### DISCUSSION TITLE
Possible improvement on sequence numbering on post deletion

### DISCUSSION THREAD

**Discussion by jeffreyvr** [Team-Tea-Time/laravel-forum#376] [Ideas]
Hi,

I have run into an issue, that when I delete a post on a thread that has a lot of posts (3000+) I get memory limit errors or:
`An exception occurred during an Action transaction.`

I'm referring to this action: https://github.com/Team-Tea-Time/laravel-forum/blob/975a1cedaf34a3ffc275f5441c63d58bc72f7e89/src/Actions/DeletePost.php

Is there a reason why the sequence number is not set directly vs. calling the method on the model. So instead:

```php
$this->post->thread->posts->each(function ($p) {
   $p->updateWithoutTouch(['sequence' => $p->getSequenceNumber()]);
});
```

to this 

```php
$this->post->thread->posts->each(function ($p, $index) {
    $p->updateWithoutTouch(['sequence' => $index + 1]);
});
```

Not having to call `getSequenceNumber` prevents the errors I'm having.

https://github.com/Team-Tea-Time/laravel-forum/blob/975a1cedaf34a3ffc275f5441c63d58bc72f7e89/src/Models/Post.php#L58

When the `getSequenceNumber` method is used, it needs to loop over the posts again. But very well might be intentional. 

I'm working with v5, though v6 does not seem to differ too much.

**Comment by Riari** [comment]
I think you're right, I can't see any reason not to make your suggested change.

Could you open a PR for it? It can target the 5.0 branch and I'll do a release once it's merged, then I'll cherry-pick it for a new 6.x release as well.
